clients/da-vinci-client/src/main/java/com/linkedin/davinci/StoreBackend.java

```diff
@@ -240,9 +243,24 @@ synchronized void trySubscribeDaVinciFutureVersion() {
     } else {
       return;
     }
-    LOGGER.info("Subscribing to future version {}", targetVersion.kafkaTopicName());
-    setDaVinciFutureVersion(new VersionBackend(backend, targetVersion, stats));
-    daVinciFutureVersion.subscribe(subscription).whenComplete((v, e) -> trySwapDaVinciCurrentVersion(e));
+
+    Store store = backend.getStoreRepository().getStoreOrThrow(storeName);
+    Set<String> targetRegions = RegionUtils.parseRegionsFilterList(store.getTargetSwapRegion());
+    String currentRegion = backend.getConfigLoader().getVeniceServerConfig().getRegionName();
+    boolean isTargetRegionEnabled = !StringUtils.isEmpty(store.getTargetSwapRegion());
```

Comment:
can we add a check to disable setting of target push using deferred swap for DVC enabled stores? I do not want to get this change deployed before all components deployed and validated.